frontend/rust-lib/flowy-database2/src/services/group/controller.rs

```diff
@@ -229,54 +284,71 @@ where
     self.context.move_group(from_group_id, to_group_id)
   }
 
-  fn did_create_row(&mut self, row: &Row, index: usize) -> Vec<GroupRowsNotificationPB> {
-    let mut changesets: Vec<GroupRowsNotificationPB> = vec![];
-
+  async fn did_create_row(
+    &mut self,
+    row: &Row,
+    index: usize,
+  ) -> Option<Vec<GroupRowsNotificationPB>> {
     let cell = match row.cells.get(&self.grouping_field_id) {
-      None => self.placeholder_cell(),
-      Some(cell) => Some(cell.clone()),
+      None => self.placeholder_cell()?,
+      Some(cell) => cell.clone(),
     };
 
-    if let Some(cell) = cell {
-      let cell_data = <T as TypeOption>::CellData::from(&cell);
+    let mut changesets: Vec<GroupRowsNotificationPB> = vec![];
+
+    let cell_data = <T as TypeOption>::CellData::from(&cell);
+
+    let mut did_insert_into_status_group = false;
+
+    let all_rows = self.delegate.get_all_rows(&self.context.view_id).await;
 
-      let mut suitable_group_ids = vec![];
+    let group_ids = self
+      .context
+      .groups()
+      .into_iter()
+      .filter(|group| group.id != self.grouping_field_id)
+      .map(|group| group.id.clone())
+      .collect::<Vec<_>>();
 
-      for group in self.get_all_groups() {
-        if self.can_group(&group.id, &cell_data) {
-          suitable_group_ids.push(group.id.clone());
-          let changeset = GroupRowsNotificationPB::insert(
-            group.id.clone(),
+    for group_id in group_ids {
+      if self.can_group(&group_id, &cell_data) {
+        // make sure that the row isn't already in the group data
+        if let Some((_, group_data)) = self.get_group(&group_id) {
+          if group_data.contains_row(&row.id) {
+            did_insert_into_status_group = true;
+            continue;
+          }
+        }
```

Comment:
@appflowy Something I found. Moving an object in an `ArrayRef` from one position to another also triggers a `Change::Added` delta, meaning that `v_did_create_row` is then called unnecessarily. I've added checks here in the group controllers, but not sure if we should do some handling at the `view_observer` level.

Reply:
yes. I am working on it